### ISSUE TITLE
coco_500 model: 'conv4_3_norm_mbox_loc' shape mismatch

### ISSUE THREAD

**Issue by minacreft** [weiliu89/caffe#149]
hi. I was trying to do some evaluation on provided coco_SSD model.
I made some changes in score_ssd_pascal.py (paths, resize_w/h, num_classes, etc), and run the script.
It works well on coco_300x300 model, while on coco_500x500 model I came across following errors:

`net.cpp:765] Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_loc'; shape mismatch.  Source param shape is 16 512 3 3 (73728); target param shape is 12 512 3 3 (55296). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.`

It seems the provided caffemodel and created prototxt are in different shapes. Where should I modify to solve this problem? Thanks!

**Comment by weiliu89** [comment]
Please refer to the deploy.prototxt for the SSD500 coco model. I might have put 4 (instead of 3) prior box on conv4_3. You need to change the prior box definition in the python script and make sure it is consistent with the provided model.

**Comment by minacreft** [comment]
Got it. Thanks !

**Comment by yanxp** [comment]
hello,I also miss the problem,can you tell how to solver the issue? thank you!

**Comment by RoseLii** [comment]
please help me
Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_conf'; shape mismatch.  Source param shape is 84 512 3 3 (387072); target param shape is 176 512 3 3 (811008). To learn this layer's parameters from scratch rather than copying from a saved net, rename the laye. @weiliu89 @minacreft

**Comment by minacreft** [comment]
in my understanding, 84 = 21 x 4, which means: numFilters = numClasses x numDefaultBoxes.
so maybe you should change 'num_classes' or 'aspect_ratios' in the python script, or rename it to train a new layer ?

**Comment by RoseLii** [comment]
@minacreft thank you.
i have a issue,please help me.
I1030 10:49:55.116433 11788 sgd_solver.cpp:138] Iteration 11360, lr = 0.001
I1030 10:50:31.269299 11788 solver.cpp:243] Iteration 11370, loss = 5.11553
I1030 10:50:31.269405 11788 solver.cpp:259]     Train net output #0: mbox_loss = 3.41015 (* 1 = 3.41015 loss)
I1030 10:50:31.269412 11788 sgd_solver.cpp:138] Iteration 11370, lr = 0.001
I1030 10:51:07.989773 11788 solver.cpp:243] Iteration 11380, loss = 5.17354
I1030 10:51:07.989881 11788 solver.cpp:259]     Train net output #0: mbox_loss = 4.46725 (* 1 = 4.46725 loss)
I1030 10:51:07.989888 11788 sgd_solver.cpp:138] Iteration 11380, lr = 0.001
I1030 10:51:40.768656 11788 solver.cpp:243] Iteration 11390, loss = 5.00373
I1030 10:51:40.768764 11788 solver.cpp:259]     Train net output #0: mbox_loss = 4.3176 (* 1 = 4.3176 loss)
I1030 10:51:40.768770 11788 sgd_solver.cpp:138] Iteration 11390, lr = 0.001
I1030 10:52:12.765962 11788 solver.cpp:243] Iteration 11400, loss = 4.99007
I1030 10:52:12.766019 11788 solver.cpp:259]     Train net output #0: mbox_loss = 5.48043 (* 1 = 5.48043 loss)
I1030 10:52:12.766026 11788 sgd_solver.cpp:138] Iteration 11400, lr = 0.001
I1030 10:52:47.248888 11788 solver.cpp:243] Iteration 11410, loss = 5.09808
I1030 10:52:47.249001 11788 solver.cpp:259]     Train net output #0: mbox_loss = 7.02698 (* 1 = 7.02698 loss)

why loss charge low and instability.

**Comment by minacreft** [comment]
sorry i know very little about training..
maybe a lower learning rate will help.

**Comment by RoseLii** [comment]
thank you very much.I will try.

**Comment by RoseLii** [comment]
@minacreft  @weiliu89 你好，就是'conv4_3_norm_mbox_conf'这个问题。我当时将num_class全都改成了我自己的类别数，还是未能解决。我在查找网上的解决方法
mbox_layers = CreateMultiBoxHead(net, data_layer='data', from_layers=mbox_source_layers,
        use_batchnorm=use_batchnorm, min_sizes=min_sizes, max_sizes=max_sizes,
        aspect_ratios=aspect_ratios, steps=steps, normalizations=normalizations,
        num_classes=num_classes, share_location=share_location, flip=flip, clip=clip,
        prior_variance=prior_variance, kernel_size=3, pad=1, lr_mult=lr_mult,conf_postfix='_kitti')
    多加了一个conf_postfix='_kitti'，训练可以了，但是加了这一句好像是将conv4-3给忽略了，但是由于我的是小目标，可能需要用到conv4-3这一层。还有什么更好的方法解决conv4_3_norm_mbox_conf'的这个问题呢。我已经训练了好久了，还是不能收敛。
求帮助

**Comment by minacreft** [comment]
我理解使用新的类别是需要重命名这些层以重新训练的。
你可以检查一下生成的 prototxt 文件，如果这些层的输入输出都正常连接了，是不会将它忽略的。
模型正常的条件下，建议再用`examples/ssd.ipynb`确认一下生成的数据集是否正常。之后就是多尝试调整超参数进行训练吧。
了解的比较浅，希望对你有所帮助～

**Comment by liangxi627** [comment]
@weiliu89 Please help me,  after changing the prior box definition in the python script, there is another error, how can I solve it? Thanks!
**F0322 19:58:32.284278  1512 sgd_solver.cpp:354] Check failed: state.history_size() == history_.size() (55 vs. 54) Incorrect length of history blobs.**
